app/routes.js

```diff
@@ -0,0 +1,173 @@
+import React from 'react';
+import { Route, IndexRoute } from 'react-router';
+
+import AppComponent from './components/app';
+import Patients from './pages/patients';
+import Login from './pages/login';
+import Signup from './pages/signup';
+import Profile from './pages/profile';
+import Patient from './pages/patient';
+import PatientNew from './pages/patientnew';
+import PatientData from './pages/patientdata';
+import RequestPasswordReset from './pages/passwordreset/request';
+import ConfirmPasswordReset from './pages/passwordreset/confirm';
+import EmailVerification from './pages/emailverification';
+
+import personUtils from './core/personutils';
+
+/**
+ * This function redirects any requests that land on pages that should only be
+ * visible when logged in if the user is logged out
+ * 
+ * @param  {Object} nextState
+ * @param  {Function} replaceState
+ *
+ * @return {boolean|null} returns true if hash mapping happened
+ */
+export const requireAuth = (api) => (nextState, replaceState) => {
+  if (!api.user.isAuthenticated()) {
+    replaceState(null, '/login');
+  }
+};
+
+/**
+ * This function redirects any requests that land on pages that should only be
+ * visible when no data storage is set up if the user has data storage set up
+ *
```

Comment:
So @GordyD here's the first of the two functions I added to deal with similar issues with routes and the sign-up flow. The first issue was an infinite loop when navigating to /patients/new (via the address bar). This is only intended to be accessible if the user hasn't "set up data storage" (== fill out profile info including diagnosis date). So I basically ended up creating this additional `onEnter` function `requireAuthAndNoPatient` to redirect to /patients if the user already *has* set up data storage.

The big downside of this approach, obviously, is that it includes redundant API calls. Not a *huge* deal, but feels a little messy and like it shouldn't be necessary. I'm hoping once we move on to redux we'll have access to the global app state from here within routes.js and won't have to make these API calls? (The logic may have to stay until we do some further refactoring, but that feels less unacceptable.)